engine/db/execution/vec_search_executor.cpp

```diff
@@ -694,32 +695,43 @@ bool VecSearchExecutor::BruteForceSearch(const float *query_data, const int64_t
     float dist = fstdistfunc_(vector_table_ + dimension_ * v_id, query_data, dist_func_param_);
     brute_force_queue_[v_id - start] = Candidate(v_id, dist, false);
   }
+
+  // compacting the result with 2 pointers by removing the deleted entries
+  // iter: the iterator to loop through the brute force queue
+  // num_result: the pointer to the current right-most slot in the new result
+  int64_t iter = 0,
+          num_result = 0;
+  // remove the invalid entries
+  for (; iter < end - start; ++iter) {
+    if (!deleted.test(iter)) {
+      if (iter != num_result) {
+        brute_force_queue_[num_result] = brute_force_queue_[iter];
+      }
+      num_result++;
+    }
+  }
+  brute_force_queue_.resize(num_result);
```

Comment:
Does frequent resize have performance problem?

Reply:
> Calling resize() with a smaller size has no effect on the capacity of a vector. It will not free memory.

https://stackoverflow.com/questions/1155693/stdvector-resize-downward